cpu/stm32/periph/spi_u5.c

```diff
@@ -0,0 +1,386 @@
+/*
+ * Copyright (C) 2014 Hamburg University of Applied Sciences
+ *               2014-2017 Freie Universität Berlin
+ *               2016-2017 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_stm32
+ * @ingroup     drivers_periph_spi
+ * @{
+ *
+ * @file
+ * @brief       Low-level SPI driver implementation
+ *
+ * @author      Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ * @author      Fabian Nack <nack@inf.fu-berlin.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ * @author      Joakim Nohlgård <joakim.nohlgard@eistec.se>
+ * @author      Thomas Eichinger <thomas.eichinger@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <assert.h>
+#include <stdint.h>
+
+#include "board.h"
+#include "compiler_hints.h"
+#include "mutex.h"
+#include "periph/cpu_gpio_ll.h"
+#include "periph/gpio.h"
+#include "periph/spi.h"
+#include "pm_layered.h"
+#include "stm32u585xx.h"
+#include "ztimer.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+/**
+ * @brief   Number of bits to shift the BR value in the CR1 register
+ */
+#define BR_SHIFT (27U)
+#define BR_MAX (7U)
+
+#define SPI_CR2_SETTINGS 0
+
+#define SPI_CFG2_SETTINGS (SPI_CFG2_AFCNTR)
+#define SPI_CFG1_SETTINGS (SPI_CFG1_CRCSIZE_Msk | (0x7 << SPI_CFG1_DSIZE_Pos))
```

Comment:
```suggestion
#define SPI_CFG1_SETTINGS ((0x7 << SPI_CFG1_DSIZE_Pos))
```

CRC mask got in there somehow?